libraries/mbed/targets/hal/TARGET_STM/TARGET_STM32F4/analogin_api.c

```diff
@@ -33,14 +33,19 @@
 #include "wait_api.h"
 #include "cmsis.h"
 #include "pinmap.h"
+#include "mbed_error.h"
 #include "PeripheralPins.h"
 
 ADC_HandleTypeDef AdcHandle;
 
-int adc_inited = 0;
-
 void analogin_init(analogin_t *obj, PinName pin)
 {
+#if defined(ADC1)
+    static int adc1_inited = 0;
```

Comment:
Hi,
I've done like this to be aligned with other platforms. We can keep it this way for the time being, and I'll check with my colleagues if this is the best way to handle the init-done flag. 
Cheers